### PR TITLE
[Docs] Fix #3128: Add api documentation for download_all_submissions API

### LINKED ISSUE #3128: Add swagger api documentation for download_all_submissions API
### Description

Hosts can use EvalAI APIs to download all submission details. The download all submissions API documentation is missing documentation on EvalAI [docs](http://eval.ai/api/docs/).

### Deliverables
Add documentation for using `download_all_submissions` API using swagger. An example of using swagger for API documentation is available [here](https://github.com/Cloud-CV/EvalAI/blob/master/apps/jobs/views.py#L87). The documentation should provide a list of query and header params described briefly and an example.

### PR DESCRIPTION
**Pull request by afif-fahim**
Fixes #3128 

After changes preview:
image
image

### CODE REVIEW

File: apps/challenges/views.py [hunk 1]
```diff
@@ -1740,14 +1740,88 @@ def get_all_submissions_of_challenge(
...
 
 
+@swagger_auto_schema(
+    methods=["get"],
+    manual_parameters=[
+        openapi.Parameter(
+            name="challenge_pk",
+            in_=openapi.IN_PATH,
+            type=openapi.TYPE_NUMBER,
+            description="Challenge ID",
```
Comment (Ram81) [inline apps/challenges/views.py:1750]: Change `Challenge ID` to `Challenge pk`

File: apps/challenges/views.py [hunk 2]
```diff
@@ -1740,14 +1740,88 @@ def get_all_submissions_of_challenge(
...
+            required=True,
+        ),
+        openapi.Parameter(
+            name="challenge_phase_pk",
+            in_=openapi.IN_PATH,
+            type=openapi.TYPE_NUMBER,
+            description="Challenge Phase ID",
```
Comment (Ram81) [inline apps/challenges/views.py:1757]: Change `Challenge Phase ID` to `Challenge phase pk`

File: apps/challenges/views.py [hunk 3]
```diff
@@ -1740,14 +1740,88 @@ def get_all_submissions_of_challenge(
...
+            required=True,
+        ),
+        openapi.Parameter(
+            name="file_type",
+            in_=openapi.IN_PATH,
+            type=openapi.TYPE_STRING,
+            description="File Type",
```
Comment (Ram81) [inline apps/challenges/views.py:1764]: Change `File Type` to `File type`

File: apps/challenges/views.py [hunk 4]
```diff
@@ -1740,14 +1740,88 @@ def get_all_submissions_of_challenge(
...
 @api_view(["GET", "POST"])
 @throttle_classes([UserRateThrottle])
 @permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
 @authentication_classes((JWTAuthentication, ExpiringTokenAuthentication))
 def download_all_submissions(
     request, challenge_pk, challenge_phase_pk, file_type
 ):
-
+    """
+    API endpoint to download all the submissions for a particular challenge as a csv
```
Comment (Ram81) [inline apps/challenges/views.py:1823]: @afif-fahim please complete the docstrings i.e mention Arguments and API response

File: apps/challenges/views.py [hunk 5]
```diff
@@ -1740,14 +1740,88 @@ def get_all_submissions_of_challenge(
...
+    },
+)
+@swagger_auto_schema(
+    methods=["post"],
+    manual_parameters=[
+        openapi.Parameter(
+            name="challenge_pk",
+            in_=openapi.IN_PATH,
+            type=openapi.TYPE_NUMBER,
+            description="Challenge ID",
```
Comment (Ram81) [inline apps/challenges/views.py:1786]: Change to `Challenge pk`

File: apps/challenges/views.py [hunk 6]
```diff
@@ -1740,14 +1740,88 @@ def get_all_submissions_of_challenge(
...
+            required=True,
+        ),
+        openapi.Parameter(
+            name="challenge_phase_pk",
+            in_=openapi.IN_PATH,
+            type=openapi.TYPE_NUMBER,
+            description="Challenge Phase ID",
```
Comment (Ram81) [inline apps/challenges/views.py:1793]: Change to `Challenge phase pk`

File: apps/challenges/views.py [hunk 7]
```diff
@@ -1740,14 +1740,88 @@ def get_all_submissions_of_challenge(
...
+            required=True,
+        ),
+        openapi.Parameter(
+            name="file_type",
+            in_=openapi.IN_PATH,
+            type=openapi.TYPE_STRING,
+            description="File Type",
```
Comment (Ram81) [inline apps/challenges/views.py:1800]: Change to `File type`

**Review by Ram81 — CHANGES_REQUESTED** [review]

**Comment by afif-fahim** [comment]
@Ram81 updated the PR

**Comment by Ram81** [comment]
@afif-fahim can you share updated screenshot of the docs page

**Comment by afif-fahim** [comment]
@Ram81, Updated screenshot of doc:
For GET:
image
For POST:
image

**Review by Ram81 — APPROVED** [review]

**Pull request merged**